### PR TITLE
SAP CF destination proxy pets-app

### PR DESCRIPTION
**Pull request by AlenaShalamberydze**

### CODE REVIEW

File: pets-proxy/src/main/java/com/leverx/pets/transaction/Transactional.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+public interface Transactional {
+
+    <T> T executeSave();
+
+    void rollback();
+
+    <T> void addEntityToResponse(UserCatDogResponse response, T entity);
+
+}
```
Comment (ilyaaleks) [inline pets-proxy/src/main/java/com/leverx/pets/transaction/Transactional.java:13]:
```
public interface Transactional<T extends Response> {

    T executeSave();

    void rollback();

    void addEntityToResponse(UserCatDogResponse response, T entity);

}
```

File: pets-proxy/src/main/java/com/leverx/pets/transaction/impl/CatTransactional.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        return catResponse;
+    }
+
+    @Override
+    public void rollback() {
+        long id = (long) currentRequestAttributes().getAttribute(CAT_ID, SCOPE_REQUEST);
+        catService.deleteById(id);
+    }
+
+    @Override
```
Comment (ilyaaleks) [inline pets-proxy/src/main/java/com/leverx/pets/transaction/impl/CatTransactional.java:41]:
```
public class CatTransactional implements Transactional<CatResponse> {

    private final CatRequest catRequest;
    private final CatService catService;

    @Override
    public CatResponse executeSave() {
        long userId = (long) currentRequestAttributes().getAttribute(USER_ID, SCOPE_REQUEST);
        catRequest.setUserId(userId);
        CatResponse catResponse = catService.save(catRequest);
        currentRequestAttributes()
                .setAttribute(CAT_ID, catResponse.getId(), SCOPE_REQUEST);
        return catResponse;
    }

    @Override
    public void rollback() {
        long id = (long) currentRequestAttributes().getAttribute(CAT_ID, SCOPE_REQUEST);
        catService.deleteById(id);
    }

    @Override
    public void addEntityToResponse(UserCatDogResponse response, CatResponse entity) {
        response.setCat(entity);
    }


}
```
Make the same with other Transactional classes
`public interface Response {}` it is marker interface

File: pets-proxy/src/main/java/com/leverx/pets/repository/implDestination/CatRepositoryDestinationImpl.java [hunk 3]
```diff
@@ -30,23 +27,16 @@
...
         CatResponse[] cats;
         try {
-            HttpEntity httpEntity = client.execute(get).getEntity();
+            HttpEntity httpEntity = destinationService.executeGetAll(CATS);
             cats = objectMapper
-                    .readValue(EntityUtils.toString(httpEntity),
+                    .readValue(EntityUtils.toString(httpEntity, CHARSET),
                             CatResponse[].class);
         } catch (IOException e) {
             log.error("Troubles getting cats from DB");
```
Comment (ilyaaleks) [inline pets-proxy/src/main/java/com/leverx/pets/repository/implDestination/CatRepositoryDestinationImpl.java]:
```
try {
            HttpEntity httpEntity = destinationService.executeGetAll(CATS);
            return asList(objectMapper
                    .readValue(EntityUtils.toString(httpEntity, org.apache.http.Consts.UTF_8),
                            CatResponse[].class));
        } catch (IOException e) {
            log.error("Troubles getting cats from DB");
            throw new RepositoryException("Troubles getting cats from DB");
        }
```
And change another methods in the same way. Also don't forget to use static import

File: pets-proxy/src/main/java/com/leverx/pets/repository/implDestination/DogRepositoryDestinationImpl.java [hunk 4]
```diff
@@ -30,23 +27,16 @@
 @Primary
```
Comment (ilyaaleks) [inline pets-proxy/src/main/java/com/leverx/pets/repository/implDestination/DogRepositoryDestinationImpl.java:27]: Read article about `@Profile`: https://dzone.com/articles/spring-boot-profiles-1
And standard profile for cf you can find there: https://github.com/cloudfoundry/java-buildpack/blob/main/docs/framework-spring_auto_reconfiguration.md
And then do not use `@Primary`

File: pets-proxy/src/main/java/com/leverx/pets/repository/implDestination/RepositoryConstants.java [hunk 5]
```diff
@@ -0,0 +1,14 @@
+package com.leverx.pets.repository.implDestination;
+
+import lombok.NoArgsConstructor;
+
+import static lombok.AccessLevel.PRIVATE;
+
+@NoArgsConstructor(access = PRIVATE)
+public final class RepositoryConstants {
```
Comment (ilyaaleks) [inline pets-proxy/src/main/java/com/leverx/pets/repository/implDestination/RepositoryConstants.java:8]: `RepositoryConstants` -> `DestinationConstants`

File: pets-proxy/src/main/java/com/leverx/pets/repository/util/UserPetRepositoryUtil.java [hunk 6]
```diff
@@ -20,11 +16,6 @@
 @NoArgsConstructor(access = PRIVATE)
 public final class UserPetRepositoryUtil {
 
-    public static HttpClient getHttpClientWithDestination(){
-        HttpDestination destination = getDestination("cf-proxy-pets").asHttp();
-        return getHttpClient(destination);
-    }
-
     public static HttpEntity<String> getHttpEntityWithoutBody(String authProvider) {
```
Comment (ilyaaleks) [inline pets-proxy/src/main/java/com/leverx/pets/repository/util/UserPetRepositoryUtil.java:19]: `authProvider` -> `authToken`

File: pets-app/src/main/java/com/example/pets/controller/AnimalController.java [hunk 7]
```diff
@@ -1,18 +1,15 @@
...
+import com.example.pets.dto.response.ResponseAnimal;
 import com.example.pets.service.AnimalService;
 import lombok.RequiredArgsConstructor;
 import org.springframework.http.ResponseEntity;
-import org.springframework.web.bind.annotation.DeleteMapping;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.PathVariable;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
```
Comment (Uladzislau-Kukoba) [inline pets-app/src/main/java/com/example/pets/controller/AnimalController.java:7]: Don't use wild card imports

**Pull request merged**